pkg/webui/components/checkbox/checkbox.js

```diff
@@ -157,25 +160,28 @@ class Checkbox extends React.PureComponent {
     })
 
     return (
-      <label className={cls}>
-        <span className={style.checkbox}>
-          <input
-            type="checkbox"
-            ref={this.input}
-            name={name}
-            readOnly={readOnly}
-            autoFocus={autoFocus}
-            onChange={this.handleChange}
-            id={id}
-            aria-describedby={rest['aria-describedby']}
-            aria-invalid={rest['aria-invalid']}
-            {...checkboxProps}
-          />
-          <span className={style.checkmark} />
-        </span>
-        {label && <Message className={style.label} content={label} />}
-        {children}
-      </label>
+      <>
+        <label className={cls}>
+          <span className={style.checkbox}>
+            <input
+              type="checkbox"
+              ref={this.input}
+              name={name}
+              readOnly={readOnly}
+              autoFocus={autoFocus}
+              onChange={this.handleChange}
+              id={id}
+              aria-describedby={rest['aria-describedby']}
+              aria-invalid={rest['aria-invalid']}
+              {...checkboxProps}
+            />
+            <span className={style.checkmark} />
+          </span>
+          {label && <Message className={style.label} content={label} />}
+          {children}
+        </label>
+        {fieldDescription && <Message className={style.description} content={fieldDescription} />}
```

Comment:
Please use the `children` prop to display the description message, which already gives us an all-purpose way to render nodes within the checkbox.

Three reasons:

*   We have been using passed messages in the past but it turned out to be not a good solution because:
    *   It's hard to pass message values (we have only a wonky workaround for that)
    *   It's hard to pass individual styling for the message
*   The prop would only serve an edge-case
*   The desired functionality can be achieved with existing props

Generally, try to consider a cost/utility ratio when deciding to add a prop or extend component functionality. Often, it cannot be avoided to extend a component but sometimes it can.